src/ring_theory/localization/basic.lean

```diff
@@ -793,6 +793,11 @@ instance : comm_semiring (localization M) :=
   right_distrib  := λ m n k, localization.induction_on₃ m n k (by tac),
   .. localization.comm_monoid_with_zero M }
 
+lemma mk_sum {ι : Type*} (f : ι → R) (s : finset ι) (b : M) :
+  mk (s.sum f) b = s.sum (λ i, mk (f i) b) :=
+by classical; exact finset.induction_on s (by simp only [finset.sum_empty, mk_zero]) (λ i s h ih,
+  by rw [finset.sum_insert h, finset.sum_insert h, ←add_mk_self, ih])
```

Comment:
Can you prove this for `list.sum` and `multiset.sum` too?

Is `mk` an additive morphism already, or possible to bundle as one? If so, you can prove this with `map_sum`.

Reply:
`flip mk b` is an add hom `R \to localization M`, but I am not sure how useful this is with rest of mathlib.

Reply:
For `list.sum`, I made two version, one use `list.foldr` and one use `(list.map _).sum`; same for `multiset`, see which version you like better.

Reply:
> `flip mk b` is an add hom `R \to localization M`, but I am not sure how useful this is with rest of mathlib.

You should add this as `mk_add_monoid_hom`, because after you do that every proof in this PR is a single line.